### ISSUE TITLE
PublishError is not public

### ISSUE THREAD

**Issue by jaskij** [nats-io/nats.rs#623]
#### NATS version (`grep 'name = "nats"' Cargo.lock -A 1`)

```
$ grep nats -A 1 Cargo.lock
name = "async-nats"
version = "0.18.0"
```

#### rustc version (`rustc --version` - we support Rust 1.41 and up)

```
$ rustc --version
rustc 1.59.0 (9d1b2106e 2022-02-23)
```

#### OS/Container environment:

```
$ cat /etc/lsb-release 
DISTRIB_ID="Arch"
DISTRIB_RELEASE="rolling"
DISTRIB_DESCRIPTION="Arch Linux"
```
Up to date

#### Steps or code to reproduce the issue:

This is an API error, but to confirm:

Create an empty crate, add a dependency on `async-nats` in `Cargo.toml`, use the code below as `main.rs`, try to compile.

```rust
struct Foo(async_nats::client::PublishError);

fn main() {}
```

#### Expected result:

`async_nats::Client::publish()` to return a public error type. In the case of the code below, `rustc` would probable suggest `PublishError` if it was public in the top-level crate.

#### Actual result:

`async_nats::Client::publish()` returns a `PublishError`, which is not public in the crate.

This error message:

```
error[E0603]: module `client` is private
   --> src/main.rs:1:24
    |
1   | struct Foo(async_nats::client::PublishError);
    |                        ^^^^^^ private module
    |
note: the module `client` is defined here
   --> /home/jaskij/.cargo/registry/src/github.com-1ecc6299db9ec823/async-nats-0.18.0/src/lib.rs:136:1
    |
136 | mod client;
    | ^^^^^^^^^^^

For more information about this error, try `rustc --explain E0603`.
error: could not compile `async-nats-test` due to previous error
```

#### Workaround

Instead of using a typed wrapper (such as enums, possibly created with `thiserror`), use an untyped wrapper (such as `Box` or `anyhow).

#### Other notes

This could have been prevented with the [`unreachable_pub`](https://doc.rust-lang.org/stable/nightly-rustc/rustc_lint/builtin/static.UNREACHABLE_PUB.html) lint.

**Comment by jaskij** [comment]
This seems relatively easy to fix, the only question being how, since there are two options:

- make the `client` module public
- `pub use client::PublishError` in `lib.rs`

**Comment by Jarema** [comment]
Thanks for the report.

We want to keep Error types close to the type that is producing them.
Will apply fix shortly.

**Comment by Jarema** [comment]
fixed in https://github.com/nats-io/nats.rs/pull/626/files

**Comment by jaskij** [comment]
Thanks for the lightning fast fix @Jarema . Any clue when it will be released?

**Comment by Jarema** [comment]
@jaskij always happy to help!

I will prepare release this week.

**Comment by Jarema** [comment]
Closing this, as it's resolved in #626 
Release coming today.
